### ISSUE TITLE
Issues with hot reloading html changes from a Razor file when @bind is used in Blazor WASM .NET 6 Preview 3

### ISSUE THREAD

**Issue by joecuevasjr** [dotnet/runtime#51855]
Using the @bind razor directive attribute breaks hot reloading HTML elements in a razor component, specifically in a Blazor WebAssembly app. 

### To Reproduce
Define a razor component with HTML and a @bind razor directive attribute
```
@page "/counter"

<h1>Counter</h1>

<p>Current count: @currentCount</p>

<button class="btn btn-primary" @onclick="IncrementCount">Click me!</button>

<input @bind="_searchTerm" />

@code {
    private string _searchTerm;

    private int currentCount = 0;

    private void IncrementCount()
    {
        currentCount++;
    }
}
```
-Run the Blazor WebAssembly application with Hot Reload.
-Change the h1 heading text in the code above, the HTML element will not update after "Hot reload of changes succeeded" message from the cli.

### Further technical details
- .NET 6.0
- SDK 6.0.100-preview.3.21202.5
- VS Community 2019 Preview 16.10.0 Preview 1.0
- Browser: Chrome

**Comment by SteveSandersonMS** [comment]
cc @lambdageek

**Comment by lewing** [comment]
@lambdageek what is the status here?

**Comment by lambdageek** [comment]
This is a `dotnet watch`+Roslyn issue.  I updated our CI infra to use Roslyns `WatchHotReloadService` and for situations like this where a lambda is in the body of a method, but the lambda contents don't change, roslyn can emit an update that Mono should be able to handle.  But for some reason deltas for Razor pages end up re-creating the lambda as a new added method (which Mono can't handle).

**Comment by lewing** [comment]
@lambdageek are we still targeting 6.0 for fixing this?

**Comment by lambdageek** [comment]
This is essentially a duplicate of https://github.com/dotnet/runtime/issues/50249

> Updating of existing lambda bodies that are generated from .razor pages will become supported once dotnet/roslyn#55407 makes it into the dotnet SDK.

That means that editing text on pages with existing `@bind` will work in .net 6 once the roslyn change lands.

> Adding new lambdas will be .net 7

Adding new `@bind`s will be in .NET 7

**Comment by ilonatommy** [comment]
@lambdageek, what is the status of this?
